### PR TITLE
Feature deckmap pretty print

### PR DESCRIPTION
**Pull request by IanLondon**
## Description:

Adapted from Artyom's #220, with `app-3-0` merged.

* ASCII rendering of `robot.deck` with `pprint` util for python command line
* HTML + CSS rendering of `robot.deck` for Jupyter (via `_repr_html_`)
* Supports custom names and shared slots

image

Check List:

- [ x] Are there breaking changes in the API and how are they being communicated? -- Nope!
- [ x] Who is reviewing your code?

### CODE REVIEW

File: api/opentrons/containers/placeable.py [hunk 1]
```diff
@@ -476,6 +476,71 @@ def has_container(self, container_instance):
         """
         return container_instance in self.containers()
 
+    def _repr_html_(self):
```
Comment (jaredgreene1) [inline api/opentrons/containers/placeable.py]: Would it be possible to place this elsewhere? It makes more sense to me to have all html pretty-printing logic centralized. Maybe this functionality could live in `util/pprint.py` as well?

**Comment by mcous** [comment]
A couple general questions for review context (sorry for my ignorance)

1. What is this feature going to be used for?
2. Is generating HTML and CSS in Python (compared to e.g. a React component/app/whatever that understands a deck object) the best way to address (1)?

File: api/opentrons/containers/placeable.py (hunk 1 above)
Comment (IanLondon) [reply api/opentrons/containers/placeable.py]: I don't love that presentation logic is inside a class in the core API, but I don't know what the alternative would be. Jupyter notebook uses an object's `_repr_html_` method to render an object in HTML, just as the python console uses `__repr__` to render an object as a string.

The `_repr_html_` method could call a util fn, but that `_repr_html_` method would still have to be part of the `Deck` class. Would that be better?

**Comment by IanLondon** [comment]
@mcous Sorry, I didn't give any context at all! @jaredgreene1 also:

This feature is for Lisa OT Protocol Writer, and Anthony Hacker, to make writing protocols a bit easier.

When you're writing a protocol for the OT1, you have to worry a lot about deck collisions. If there are 2 pipettes, one pipette can crash into a container as the other goes down to interact with the container to the left/right of it. So what people end up doing is sketching the deck layout in a notebook as they go.

Putting this in Jupyter notebook is a quick win. When Luisa + Jerry write protocols, they use Jupyter to iteratively test the outputs. I can't find a simple, maintainable way to do it with React-in-Jupyter. I think the long-term might be having these protocol visualizations alongside the "write new protocol" tool in the new app, instead of in Jupyter notebook - but that seems very far off.

File: api/opentrons/containers/placeable.py (hunk 1 above)
Comment (jaredgreene1) [reply api/opentrons/containers/placeable.py]: Oohh, didn't realize that it was an implicit call.... Ehh, I think it's fine in the class then.

**Pull request closed without merging**